src/Raven.Server/Documents/PeriodicBackup/BackupTask.cs

```diff
@@ -737,29 +736,17 @@ private void AddInfo(string message)
 
         public Stream GetOutputStream(Stream fileStream)
         {
-            if ((_database.MasterKey == null) &&
-                (_configuration.BackupEncryptionSettings == null))
-                return fileStream;
-
-            if ((_database.MasterKey == null) &&
-                (_configuration.BackupEncryptionSettings?.EncryptionMode == EncryptionMode.None))
-                return fileStream;
-
-            if ((_database.MasterKey != null) &&
-                (_configuration.BackupEncryptionSettings?.EncryptionMode == EncryptionMode.None))
+            if (_isBackupEncrypted == false)
                 return fileStream;
 
-            if ((_database.MasterKey != null) && (_configuration?.BackupEncryptionSettings == null))
-                return new EncryptingXChaCha20Poly1305Stream(fileStream,
-                    _database.MasterKey);
+            if (_database.MasterKey != null && _configuration.BackupEncryptionSettings == null)
+                return new EncryptingXChaCha20Poly1305Stream(fileStream, _database.MasterKey);
 
-            if (_configuration?.BackupEncryptionSettings?.EncryptionMode == EncryptionMode.UseDatabaseKey)
-                return new EncryptingXChaCha20Poly1305Stream(fileStream,
-                    _database.MasterKey);
+            if (_configuration.BackupEncryptionSettings.EncryptionMode == EncryptionMode.UseDatabaseKey)
+                return new EncryptingXChaCha20Poly1305Stream(fileStream, _database.MasterKey);
```

Comment:
if `_database.MasterKey` is null we need to throw?

Reply:
It's validated before the backup starts:
https://github.com/grisha-kotler/ravendb/blob/bfeb644d706ca5e844c8f75dc474298a2d99f0c5/src/Raven.Server/Documents/PeriodicBackup/BackupTask.cs#L645